pkg/alertmanager/amcfg.go

```diff
@@ -121,6 +127,59 @@ func (cg *configGenerator) generateConfig(
 	baseConfig.Route.Routes = append(subRoutes, baseConfig.Route.Routes...)
 
 	return yaml.Marshal(baseConfig)
+
+}
+
+func convertTimeInterval(in *monitoringv1alpha1.MuteTimeInterval) *MuteTimeInterval {
+	var mti MuteTimeInterval
+
+	mti.Name = in.Name
+
+	for _, interval := range in.TimeInterval {
+
+		times := make([]TimeRange, len(interval.Times))
+		days := make([]string, len(interval.DaysOfMonth))
+		months := make([]string, len(interval.Months))
+		week := make([]string, len(interval.Weekdays))
+		years := make([]string, len(interval.Years))
+
+		for j, y := range interval.Times {
+
+			times[j].StartTime = y.StartTime
+			times[j].EndTime = y.EndTime
+		}
+
+		for j, y := range interval.DaysOfMonth {
+			days[j] = y
+		}
+
+		for j, y := range interval.Months {
+			months[j] = y
+		}
+
+		for j, y := range interval.Weekdays {
+			week[j] = y
+		}
+
+		for j, y := range interval.Years {
+			years[j] = y
+		}
+
+		x := &TimeInterval{
+			Times:       times,
+			DaysOfMonth: days,
+			Months:      months,
+			Weekdays:    week,
+			Years:       years,
+		}
+
+		// TODO, fix this hack to prevent a empty object being added to the yaml
```

Comment:
Is this TODO something you intended address in the current PR?

Reply:
Updated comment.